Cargo.toml

```diff
@@ -14,5 +14,34 @@ members = [
   "burn-tch",
   "burn-ndarray",
   "burn-autodiff",
+  "burn-common",
+  "burn-no-std-tests",
   "examples/*",
 ]
+
+[workspace.dependencies]
+cfg-if = "1.0.0"
+const-random = "0.1.15"
+dashmap = "5.4.0"
+derive-new = {version = "0.5.9", default-features = false}# Default enables std   
+dirs = "4.0.0"
+fake = "2.5.0"
+flate2 = "1.0.25"
+half = {version = "1.6.0", features = [
+  "num-traits",
+]}# needs to be 1.6 to work with tch
+hashbrown = "0.13.2"
+indicatif = "0.17.2"
+lazy_static = "1.4.0"
+libm = "0.2.6"
+log = "0.4.17"
+log4rs = "1.2.0"
+ndarray = {version = "0.15.6", default-features = false}# Default enables std        
+num-traits = {version = "0.2.15", default-features = false, features = ["libm"]}# Default enables std 
+rand = {version = "0.8.5", default-features = false, features = ["std_rng"]}# Default enables std   
+rand_distr = {version = "0.4.3", default-features = false}# Default enables std   
+serde = {version = "1.0.152", default-features = false, features = ["derive", "alloc"]}# Default enables std   
+serde_json = {version = "1.0.91", default-features = false}# Default enables std   
+spin = {version = "0.9.5", features = ["mutex", "spin_mutex"]}
+thiserror = "1.0.38"
+uuid = {version = "1.3.0", default-features = false}
```

Comment:
I would separate the dependencies in blocks with comments about why it is need and maybe features comments comments.

Reply:
OK. Agree. Done.